sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -591,6 +591,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
       children.last.generateTreeString(
         depth + 1, lastChildren :+ true, append, verbose, prefix, addSuffix, maxFields)
     }
+    append("")
```

Comment:
Are you using the boolean returned by the append for control flow anywhere?

Let's remove it for the initial PR, we can always address this in  a follow-up.

Reply:
Moreover, if you are appending empty strings all over the place then it would be good to make `append` drop these.